ctru-rs/src/services/ac.rs

```diff
@@ -0,0 +1,349 @@
+        }
+    }
+
+    /// Returns whether the console is connected to Wi-Fi
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    ///
+    /// println!("Wi-Fi connected: {}", ac.get_wifi_status()?);
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "ACU_GetWifiStatus")]
+    pub fn get_wifi_status(&self) -> crate::Result<bool> {
+        unsafe {
+            let mut ret = 0u32;
+            ResultCode(ctru_sys::ACU_GetStatus(&mut ret))?;
+
+            Ok(ret == 3)
+        }
+    }
+
+    /// Returns the [SecurityMode] of the currently connected network, or error if the console isn't connected to any network.
+    ///
+    /// You can check if the console is connected to a network using [`Ac::get_wifi_status()`]
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    ///
+    /// if ac.get_wifi_status()? {
+    ///     println!("Network security: {:?}");
+    /// }
+    ///
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "ACU_GetWifiSecurityMode")]
+    pub fn get_wifi_security(&self) -> crate::Result<SecurityMode> {
+        unsafe {
+            let mut ret = 0u32;
+            ResultCode(ctru_sys::ACU_GetSecurityMode(&mut ret))?;
+            // fix this, for some reason the bindings have the type as a struct and not enum
+            // and so i can't impl TryFrom automatically
+            Ok(std::mem::transmute(ret))
+        }
+    }
+
+    /// Returns the SSID of the Wi-Fi network the console is connected to, or error if the console isn't connected to any network.
+    ///
+    /// You can check if the console is connected to a network using [`Ac::get_wifi_status()`]
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    ///
+    /// println!("The console is connected to the network \"{}\"", ac.get_wifi_ssid().unwrap());
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "ACU_GetSSID")]
+    pub fn get_wifi_ssid(&self) -> crate::Result<String> {
+        unsafe {
+            let mut len = 0u32;
+            ResultCode(ctru_sys::ACU_GetSSIDLength(&mut len))?;
+            // we don't really need space for the terminator
+            let mut vec = vec![0u8; len as usize];
+            ResultCode(ctru_sys::ACU_GetSSID(vec.as_mut_ptr()))?;
+            Ok(String::from_utf8(vec)?)
```

Comment:
Rather than returning the `String`'s error, it might be wiser to check for `Ac::get_wifi_status()` and return an error based on *that*.

It might also give us the chance to avoid implementing `Error::Utf8`.